imu_filter_madgwick/src/imu_filter.cpp

```diff
@@ -540,4 +560,11 @@ void ImuFilter::reconfigCallback(FilterConfig& config, uint32_t level)
   ROS_INFO("Magnetometer bias values: %f %f %f", mag_bias_.x, mag_bias_.y, mag_bias_.z);
 }
 
-
+void ImuFilter::imuMagVectorCallback(const MagVectorMsg::ConstPtr& mag_vector_msg)
+{
+  MagMsg mag_msg;
+  mag_msg.header = mag_vector_msg->header;
+  mag_msg.magnetic_field = mag_vector_msg->vector;
+  mag_msg.magnetic_field_covariance[0] = -1;
```

Comment:
This line should be deleted: The MagneticField.msg specifies that...

> A covariance matrix of all zeros will be interpreted as "covariance unknown"
> and to use the data a covariance will have to be assumed or gotten from some
> other source

... which is what we want. You probably mixed this up with Imu.msg, where -1 means "orientation is invalid".